### ISSUE TITLE
[TASK] Implement PredictMasked (BERT-like masking)

### ISSUE THREAD

**Issue by marcromeyn** [NVIDIA-Merlin/models#692]
# BERT-like masking

Let’s say we have a sequence `ABCDE`, BERT-like masking would result in the following:
| **Inputs**  | A        | `MASKED` | C        | `MASKED` | E        |
| ------- | -------- | -------- | -------- | -------- | -------- |
| **Targets** | `MASKED` | B        | `MASKED` | D        | `MASKED` |

_Note, the number of target items might differ for different samples in the batch. We should ensure that we have at least one target and at most `len(seq) - 1`_

The class that’s responsible for masking could roughly look like:

```python
class PredictMasked(DataAugmentation):
	def __init__(
		self, 
		schema: Schema
		target: Union[str, Tag, ColSchema],
		prediction_block=None,
		mask_selection_rate,
		mask_selection_length,
		unselectable_token_ids=[0],
		mask_token_rate=0.8,
		random_token_rate=0.1
	):
		...

	def compute_mask(self, ...):
		...
```

We want to make use of standard Keras functionality w.r.t. masking. Some useful links:
- [Mask propagation](https://keras.io/guides/understanding_masking_and_padding/)
- [MLMMaskGenerator](https://github.com/keras-team/keras-nlp/blob/master/keras_nlp/layers/mlm_mask_generator.py#L20) in [keras_nlp](https://github.com/keras-team/keras-nlp)
	- Works on normal + `tf.RaggedTensor`s
	- Uses `tensorflow_text`
- [mask_language_model](https://github.com/tensorflow/text/blob/master/tensorflow_text/python/ops/masking_ops.py#L91) in [tensorflow_text](https://github.com/tensorflow/text)
